gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidatorTest.java

```diff
@@ -0,0 +1,111 @@
+                                             .setStopId("1001")
+                                             .setStopSequence(1)
+                                             .setShapeDistTraveled(1.0)
+                                             .build();
+  private final GtfsStopTime stopTime2 = new GtfsStopTime.Builder()
+                                             .setCsvRowNumber(2)
+                                             .setTripId("trip1")
+                                             .setStopId("1002")
+                                             .setStopSequence(2)
+                                             .setShapeDistTraveled(1.5)
+                                             .build();
+  private final GtfsStopTime stopTime3 = new GtfsStopTime.Builder()
+                                             .setCsvRowNumber(3)
+                                             .setTripId("trip1")
+                                             .setStopId("1003")
+                                             .setStopSequence(3)
+                                             .setShapeDistTraveled(1.4)
+                                             .build();
+  private final GtfsStopTime stopTime4 = new GtfsStopTime.Builder()
+                                             .setCsvRowNumber(3)
+                                             .setTripId("trip1")
+                                             .setStopId("1003")
+                                             .setStopSequence(3)
+                                             .setShapeDistTraveled(1.5)
+                                             .build();
+
+  @Test
+  public void stopTimeIncreasingDistanceShouldNotGenerateNotice() {
+    final NoticeContainer noticeContainer = new NoticeContainer();
+    final StopTimeIncreasingDistanceValidator validator = new StopTimeIncreasingDistanceValidator();
+
+    // Create stopTimeTable:
+    final List<GtfsStopTime> stopTimes = new ArrayList<>();
+    stopTimes.add(stopTime1);
+    stopTimes.add(stopTime2);
+    validator.table = GtfsStopTimeTableContainer.forEntities(stopTimes, noticeContainer);
+
+    validator.validate(noticeContainer);
+    assertThat(noticeContainer.getNotices()).isEmpty();
+  }
+
+  @Test
+  public void stopTimeDecreasingDistanceShouldGenerateNotice() {
+    final NoticeContainer noticeContainer = new NoticeContainer();
+    final StopTimeIncreasingDistanceValidator validator = new StopTimeIncreasingDistanceValidator();
+
+    // Create stopTimeTable:
+    final List<GtfsStopTime> stopTimes = new ArrayList<>();
+    stopTimes.add(stopTime1);
+    stopTimes.add(stopTime2);
+    stopTimes.add(stopTime3);
+    validator.table = GtfsStopTimeTableContainer.forEntities(stopTimes, noticeContainer);
+
+    validator.validate(noticeContainer);
+    assertThat(noticeContainer.getNotices())
+        .containsExactly(new DecreasingStopTimeDistanceNotice("trip1", 3, 3, 1.4, 2, 2, 1.5));
```

Comment:
Please either update DecreasingStopTimeDistanceNotice to have a builder and use it, or improve readability by commenting at the call site to indicate which param you are setting.

For example:
DecreasingStopTimeDistanceNotice(/* tripId= */ "trip1", /* csvRowNumber = */ 3, /* stopSequence = */ 3, /* shapeDistTraveled = */ 1.4, /* prevCsvRowNumber= */ 2, /* prevStopSequence= */ 2, /* prevShapeDistTraveled = */ 1.5)

Reply:
Updated to comment at the call site